reactive/containerd.py

```diff
@@ -421,6 +421,12 @@ def upgrade_charm():
     # Prevent containerd apt pkg from being implicitly updated.
     apt_hold(CONTAINERD_PACKAGE)
 
+    clear_flag("containerd.resource.installed")
+    if DB.get("containerd-resource-overridden"):
+        # if a resource is currently overriding the deb,
+        # upgrade containerd from the apt packages
+        reinstall_containerd()
+
```

Comment:
charm-upgrade fires when you upgrade the charm (naturally) or attach a resource.  So if the user attaches a new containerd resource, we should clear the `containerd.resource.installed` flag.  If the resource was PREVIOUSLY overriding the deb sources, we'll need to first install the deb sources and then react to any other flag changes.